Week 04/Lecture 07/Assignment 02/assignment_code/src/main/java/com/example/assignment_code/service/EmployeeServiceConstructor.java

```diff
@@ -0,0 +1,19 @@
+package com.example.assignment_code.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+
+@Service
+public class EmployeeServiceConstructor {
+    private final EmailService emailService;
+
+    @Autowired
```

Comment:
Can we remove @Autowire here? Pls explain

Reply:
Yes we can, in this case, we can remove the `@ Autowired` annotation from the construction. In Spring, when a class has a `single constructor`, the Spring will automatically use that constructor for dependency injection even if the `@ Autowired` annotation is removed.

Spring automatically resolve and inject dependencies if there is only one constructor or `single constructor`. But, if there is multiple constructor, it need to give the annotation to one of them with `@ Autowired` to tell Spring which one to use.

I also add this answer and the question in the `Additional` section on assignment 2 - lecture 7 - week 4